server/src/socketHandlers/socketHandlers.ts

```diff
@@ -177,19 +177,45 @@ export default function initializeSocketHandler(io: Server) {
     });
 
     socket.on("start_game", ({ lobbyId, members }) => {
-      // Instantiate relevant classes
-      //const gameCards = initializeGameCards(members);
-      //const playerObjects = initializePlayers(members, gameCards);
-      //const dice = new Dice(SixSidedDie);
-      //game = new QwixxLogic(playerObjects, dice);
       lobbiesMap[lobbyId].startGame();
 
-      // Create path data and players' gameboard states to send back to client
-      const initialPlayersState = lobbiesMap[lobbyId].gameLogic?.players;
+      // Create path data and serialized gamelogic object to the front end
+      const gameLogic = lobbiesMap[lobbyId].gameLogic;
+      const serializedGameLogic = gameLogic ? gameLogic.serialize() : null;
       const path = `/game/${lobbyId}`;
-      const responseData = { path: path, players: initialPlayersState };
+      const responseData = { path: path, gameState: serializedGameLogic };
 
       io.to(lobbyId).emit("game_initialised", responseData);
+      console.log(serializedGameLogic);
+    });
+
+    socket.on("mark_numbers", ({ lobbyId, userId, playerChoice }) => {
+      const gameLogic = lobbiesMap[lobbyId].gameLogic;
+
+      if (!gameLogic) {
+        console.log(`Game logic not found for lobby ${lobbyId}`);
+        return;
+      }
+
+      const player = gameLogic.players.find((player) => player.name === userId);
+
+      if (!player) {
+        console.log(`Player ${userId} not found in lobby`);
+        return;
+      }
+
+      console.log(playerChoice);
+
+      const { row: rowColour, num } = playerChoice;
+      player.gameCard.markNumbers(rowColour, num);
+      console.log(player.gameCard.MarkedNumbers);
+
+      const serializedGameLogic = gameLogic ? gameLogic.serialize() : null;
```

Comment:
The logic behind the code is correct, but I think that it is now tightly coupled to the implementation of the classes. Ideally, it should just receive the "({lobbyId, userId, playerChoice})" as you have written and then call a method from the lobby class or the gameLogic class that does all this. For example, if you look at the game logic class, there is already a make move method that should handle it. It might need updating. 

That way, this socket.on() only needs to know that it calls the makeMove method and doesn't need to care about how it does it. 

So up till line 198 is probably ok, but after that you could do something like 

```
const result = gameLogic.makeMove(userId, playerChoice)
io.to(lobbyId).emit("update_markedNumbers, result);
```

Also line 213 is something we have to weight the pros and cons about. I think doing it how you've written would means that there would be no additional logic needed on client side as it's sending back the up to date state. The downside is that you're sending more data. The alternative is to send back only the part that has any changes. So {<playerId>: {red: 5}} as an example. It's sending back minimal data, but then we would need to implement the logic to update the state of the game in the client as well.

Reply:
Sorry I just looked over the code again. I didnt realise the mark number method took 3 parameters. Yeah I can update this code. Yeah I think we discussed for the purposes of the code we have it should be ok to send back the entire gamestate rather than sending individual changes and then the front end updating state.